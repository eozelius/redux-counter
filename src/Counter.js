import React, { Component } from 'react'
import Button from './shared/Button'
import { incrementCounter, decrementCounter } from './actions.js'
import { connect } from 'react-redux'


class Counter extends Component {
  render(){
    return (
      <div className='counter'>
        <div className='count'>
          <h2>Counter</h2>
          {this.props.count}
        </div>

        <div className='mutate-count'>
          <Button className='increment-btn' handleClick={this.props.increment}>Increment</Button>
          <Button className='decrement-btn' handleClick={this.props.decrement}>Decrement</Button>
        </div>
      </div>
    )
  }
}

const mapStateToProps = state => {
  return {
    count: state.count
  }
}

const mapDispatchToProps = dispatch => {
  return {
    increment: () => dispatch(incrementCounter()),
    decrement: () => dispatch(decrementCounter())
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Counter)
