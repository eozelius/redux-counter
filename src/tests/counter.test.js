import React from 'react'
import { Counter } from '../Counter'
// import configureStore from '../configureStore'
// import ConnectedCounter from '../Counter'

describe(Counter, () => {
  it('renders properly', () => {
    const counter = shallow(<Counter  />)
    expect(counter).toMatchSnapshot()
  })

  it('has an initial count of 0', () => {
    const counter = mount(<Counter count={0} />)
    expect(counter).toMatchSnapshot()
    expect(counter.prop('count')).toEqual(0)
  })

  // it.only('maps the "increment()" dispatch to props', () => {
  //   const store = configureStore()
  //   const counter = mount(<ConnectedCounter store={store} />)
  //
  //   // Definitely room for improvement.  I need to figure out how to inspect the dispatches that are being mapped to props.  Right now I only know that props.dispatch is a function.
  //   expect(counter.prop('store')['dispatch']).not.toEqual(undefined)
  // })
})
