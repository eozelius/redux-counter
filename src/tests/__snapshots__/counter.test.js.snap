// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter has an initial count of 0 1`] = `
<Counter
  count={0}
>
  <div
    className="counter"
  >
    <div
      className="count"
    >
      <h2>
        Counter
      </h2>
      0
    </div>
    <div
      className="mutate-count"
    >
      <Button
        className="increment-btn"
      >
        <button
          style={
            Object {
              "margin": "5%",
            }
          }
        >
          Increment
        </button>
      </Button>
      <Button
        className="decrement-btn"
      >
        <button
          style={
            Object {
              "margin": "5%",
            }
          }
        >
          Decrement
        </button>
      </Button>
      <Button
        className="async-increment"
      >
        <button
          style={
            Object {
              "margin": "5%",
            }
          }
        >
          Async Increment
        </button>
      </Button>
      <Button
        className="async-decrement"
      >
        <button
          style={
            Object {
              "margin": "5%",
            }
          }
        >
          Async Decrement
        </button>
      </Button>
    </div>
  </div>
</Counter>
`;

exports[`Counter renders properly 1`] = `
<div
  className="counter"
>
  <div
    className="count"
  >
    <h2>
      Counter
    </h2>
  </div>
  <div
    className="mutate-count"
  >
    <Button
      className="increment-btn"
    >
      Increment
    </Button>
    <Button
      className="decrement-btn"
    >
      Decrement
    </Button>
    <Button
      className="async-increment"
    >
      Async Increment
    </Button>
    <Button
      className="async-decrement"
    >
      Async Decrement
    </Button>
  </div>
</div>
`;
